include/solvers/fluid_dynamics_matrix_free.h

```diff
@@ -27,22 +27,31 @@ using namespace dealii;
 template <typename VectorType>
 class PreconditionBase;
 
+template <typename VectorType, typename VectorTypePrecondition>
+class PreconditionAdapter;
+
 /**
  * @brief A geometric multigrid preconditioner compatible with the
  * matrix-free solver.
  */
 template <int dim>
 class MFNavierStokesPreconditionGMG
 {
-  using VectorType     = LinearAlgebra::distributed::Vector<double>;
-  using LSTransferType = MGTransferMatrixFree<dim, double>;
-  using GCTransferType = MGTransferGlobalCoarsening<dim, VectorType>;
-  using OperatorType   = NavierStokesOperatorBase<dim, double>;
-  using SmootherPreconditionerType = PreconditionBase<VectorType>;
+  using Number             = double;
+  using MGNumber           = float;
```

Comment:
The place to switch between `float` and `double`.

Reply:
We might need a compiler flag to change this accordingly?

Reply:
I am not sure how  a compiler flag would help. Switching between float and double directly in the code could seems to be easy enough.